lib/pure/strutils.nim

```diff
@@ -1333,30 +1333,53 @@ proc findAux(s, sub: string, start: int, a: SkipTable): int =
     inc(j, a[s[j+m]])
   return -1
 
-proc find*(s, sub: string, start: Natural = 0): int {.noSideEffect,
+when not (defined(nimvm) or defined(js) or defined(nimdoc) or defined(nimscript)):
+  proc c_memchr(cstr: pointer, c: char, n: csize): pointer {.
+                importc: "memchr", header: "<string.h>" .}
+  const hasCStringBuiltin = true
+else:
+  const hasCStringBuiltin = false
+
+proc find*(s, sub: string, start: Natural = 0, last: Natural = 0): int {.noSideEffect,
```

Comment:
Hey,
I wouldn't say it require a last parameter, but it very useful in some cases.
Say, I read in a very large string to use as a cache.
I can use strutils.find(cache, start, last) to find a char/string/set[char] within those bounds.
I can do this for multiple sets of bounds without searching the whole cache miltiple time.
My use current use-case is MO file format which store strings as NULL terminated string inside a large string cache.